plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -0,0 +1,114 @@
+      PlatformUI.getWorkbench().getSharedImages().getImageDescriptor(ISharedImages.IMG_OBJS_ERROR_TSK).createImage();
+
+  private DeployPreferencesPanel content;
+  private IProject project;
+
+  private CLabel validationMessage;
+
+  public DeployPreferencesDialog(Shell parentShell, IProject project) {
+    super(parentShell);
+    this.project = project;
+  }
+
+  @Override
+  protected void configureShell(Shell newShell) {
+    super.configureShell(newShell);
+    newShell.setText("Deployment preferences");
+  }
+
+  @Override
+  protected Control createDialogArea(final Composite parent) {
+    validationMessage = new CLabel(parent, SWT.LEFT);
+    GridData layoutData = new GridData(SWT.LEFT, SWT.CENTER, true, false);
+    layoutData.exclude = true;
+    validationMessage.setLayoutData(layoutData);
+    validationMessage.setVisible(false);
+
+    Composite dialogArea = (Composite) super.createDialogArea(parent);
+    ScrolledComposite scrolledComposite = new ScrolledComposite(dialogArea, SWT.V_SCROLL | SWT.H_SCROLL);
+    scrolledComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+    scrolledComposite.setShowFocusedControl(true);
+    scrolledComposite.setExpandHorizontal(true);
+    scrolledComposite.setExpandVertical(true);
+    scrolledComposite.addControlListener(new ControlListener() {
+      @Override
+      public void controlResized(ControlEvent e) {
+        Point computeSize = DeployPreferencesDialog.this.getShell().computeSize(SWT.DEFAULT, SWT.DEFAULT);
+        DeployPreferencesDialog.this.getShell().setSize(computeSize);
+      }
+
+      @Override
+      public void controlMoved(ControlEvent e) {
+      }
+    });
+
+    content = new DeployPreferencesPanel(scrolledComposite, project);
+    content.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+    scrolledComposite.setContent(content);
+
+    DataBindingContext dataBindingContext = content.getDataBindingContext();
+    final AggregateValidationStatus validationStatus =
+        new AggregateValidationStatus(dataBindingContext, AggregateValidationStatus.MAX_SEVERITY);
+    validationStatus.addValueChangeListener(new IValueChangeListener() {
+      @Override
+      public void handleValueChange(ValueChangeEvent event) {
+        IStatus status = (IStatus) validationStatus.getValue();
+        if (status.isOK()) {
+          updateValidationStatus("");
+        } else {
+          updateValidationStatus(status.getMessage());
+        }
+      }
+    });
+    return content;
+  }
+  
+  
+  private void updateValidationStatus(String errorMessage) {
+    Preconditions.checkNotNull(errorMessage, "errorMessage is null");
+    
+    if (!validationMessage.isDisposed()) {
+      getButton(IDialogConstants.OK_ID).setEnabled(errorMessage.isEmpty());
+      validationMessage.setText(errorMessage);
+      validationMessage.setImage(errorMessage.isEmpty() ? null : ERROR_ICON);
+      validationMessage.setVisible(!errorMessage.isEmpty());
```

Comment:
If we are hiding/showing this control, maybe we could remove the call `setImage()` right above?